### PR TITLE
Fixes from stu3 branch

### PR DESCRIPTION
**Pull request by cmoesel**
Brings over some of the good work that's been done on the STU3 branches.

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -15,6 +17,16 @@ func main() {
 		s.Engine.Use(server.RequestLoggerHandler)
 	}
 
+	s.Engine.Use(cors.Middleware(cors.Config{
+		Origins:         "*",
+		Methods:         "GET, PUT, POST, DELETE",
+		RequestHeaders:  "Origin, Authorization, Content-Type",
```
Comment (eedrummer) [inline server.go]: Consider adding "If-Match" and "If-None-Exist". I don't think we support these to do conditional updates or creates, but just something to keep in mind.

File: server.go [hunk 2]
```diff
@@ -15,6 +17,16 @@ func main() {
...
+		ExposedHeaders:  "",
```
Comment (eedrummer) [inline server.go]: Maybe add "ETag" and "Last-Modified". I don't think we create these either, but just something to keep in mind.

**Comment by eedrummer** [comment]
resource_controller contains: `c.Header("Access-Control-Allow-Origin", "*")` in a bunch of handlers for really primitive CORS support previous to this PR. Those should be removed.

File: server.go (hunk 1 above)
Comment (cmoesel) [inline server.go]: Just to be sure we're on the same page, add them to `RequestHeaders`?

File: server.go (hunk 2 above)
Comment (cmoesel) [inline server.go]: To `ExposedHeaders`, right?

**Comment by cmoesel** [comment]
Yeah, I realized that this morning too.  Thanks for pointing it out.

File: server.go (hunk 2 above)
Comment (eedrummer) [inline server.go]: Yup

File: server.go (hunk 1 above)
Comment (eedrummer) [inline server.go]: Yes.

**Pull request merged**